src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -40,6 +41,28 @@ internal sealed class CircuitStateController<T> : IDisposable
         _telemetry = telemetry;
     }
 
+    public CircuitStateController(
```

Comment:
This seems quite duplicative, rather than just adding a new parameter to the existing constructor.

Reply:
Agree, let's just introduce new `Func<BreakDurationGeneratorArguments, ValueTask<TimeSpan>>? breakDurationGenerator` argument to existing constructor.